lib/pdn/Kconfig

```diff
@@ -29,6 +29,7 @@ config PDN_INIT_PRIORITY
 
 config PDN_DEFAULTS_OVERRIDE
 	bool "Override defaults for PDP context 0"
+	depends on LTE_LINK_CONTROL
```

Comment:
I'd not to see this. 
Since libmodem init callback can still config PDN, there is definite need of dependency on lib_lte_lc.